src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -530,56 +584,105 @@ impl<'tcx> Tree {
         &mut self,
         access_kind: AccessKind,
         tag: BorTag,
-        access_range: AllocRange,
+        access_range: Option<AllocRange>,
         global: &GlobalState,
         span: Span,                             // diagnostics
         access_cause: diagnostics::AccessCause, // diagnostics
     ) -> InterpResult<'tcx> {
-        for (perms_range, perms) in self.rperms.iter_mut(access_range.start, access_range.size) {
-            TreeVisitor { nodes: &mut self.nodes, tag_mapping: &self.tag_mapping, perms }
-                .traverse_parents_this_children_others(
-                    tag,
-                    |args: NodeAppArgs<'_>| -> Result<ContinueTraversal, TransitionError> {
-                        let NodeAppArgs { node, mut perm, rel_pos } = args;
+        use std::ops::Range;
+        // Per-node application:
+        // - insert the permission if it does not exist
+        // - perform the access
+        // - record the transition
+        // to which some optimizations are added:
+        // - skip the traversal of the children in some cases
+        // - do not record noop transitions
+        //
+        // This needs some extra data: the range of the `RangeMap`
+        // (the one that is not visible in diagnostics, only used to record events)
+        // is known only at call site.
```

Comment:
I don't understand this sentence. The range is known only at the call site? The call site of what? Why do you care?

Are you explaining why this is a closure? I don't think that's necessary. It took me a good minute to even realize that this might be what you are explaining here.^^

This is a closure, so the comment should be written like the doc comment for a `fn`: say what it does and what the arguments do.